{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**TREAT**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fastas = glob.glob(\"/home/tiago/documentos/github/ARG-Sentinel/genomes_hc/\" + \"*.fasta\")\n",
    "for file in fastas:\n",
    "\tfor record in SeqIO.parse(f\"{file}\",\"fasta\"):\n",
    "\t\tif len(record.seq) > 200:\n",
    "\t\t\tnew_file = open(f\"treated_{file.split('/')[-1]}\",\"a\")\n",
    "\t\t\tnew_file.write(f\"{record.id}\\n\")\n",
    "\t\t\tnew_file.write(f\"{record.seq}\\n\")\n",
    "\t\t\tnew_file.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**GET INTO**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "files = glob.glob(\"/home/tiago/documentos/github/ARG-Sentinel/suberror/*.txt\")\n",
    "genome = []\n",
    "node = []\n",
    "start = []\n",
    "end = []\n",
    "\n",
    "for f in files:\n",
    "    txt = open(f,'r')\n",
    "    for lines in txt:\n",
    "        if lines.startswith(\"NODE\"):\n",
    "            genome.append(f.split(\"/\")[-1].split(\".\")[0])\n",
    "            \n",
    "            node.append(lines.split(\"\\t\")[0])\n",
    "                        \n",
    "            start.append(lines.split(\"\\t\")[2].split(\"..\")[0])\n",
    "                                      \n",
    "            end.append(lines.split(\"\\t\")[2].split(\"..\")[1])\n",
    "                        \n",
    "                        \n",
    "df = pd.DataFrame({\"genome\":genome, \"node\":node, \"start\":start, \"end\":end})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**TRIM**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"error_info.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = [\"NODE_22_length_73939_cov_13.042389\", \n",
    "     \"NODE_37_length_43294_cov_12.460632\", \n",
    "     \"NODE_42_length_42978_cov_16.990438\", \n",
    "     \"NODE_15_length_136795_cov_22.099162\",\n",
    "     \"NODE_34_length_63067_cov_25.930044\",\n",
    "     \"NODE_44_length_43061_cov_24.463410\",\n",
    "     \"NODE_56_length_2305_cov_620.290118\",\n",
    "     \"NODE_41_length_4753_cov_20.978728\",\n",
    "     \"NODE_27_length_10506_cov_41.427597\",\n",
    "     \"NODE_43_length_2345_cov_2403.472395\",\n",
    "     \"NODE_18_length_24824_cov_61.573873\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "fasta = glob.glob(\"/home/tiago/documentos/github/ARG-Sentinel/genomes_hc/*.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "for files in fasta:\n",
    "    for record in SeqIO.parse(files,'fasta'):\n",
    "        for tag in n:\n",
    "            if record.id == tag:\n",
    "                file = open(f\"/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/{record.id}.fasta\",\"w\")\n",
    "                file.write(f\">{record.id}\\n\")\n",
    "                file.write(f\"{record.seq}\\n\")\n",
    "                file.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**PROKKA**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:07:09] This is prokka 1.13\n",
      "[20:07:09] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:07:09] Homepage is https://github.com/tseemann/prokka\n",
      "[20:07:09] Local time is Wed Jun 24 20:07:09 2020\n",
      "[20:07:09] You are tiago\n",
      "[20:07:09] Operating system is linux\n",
      "[20:07:09] You have BioPerl 1.007002\n",
      "[20:07:09] System has 12 cores.\n",
      "[20:07:09] Will use maximum of 8 cores.\n",
      "[20:07:09] Annotating as >>> Bacteria <<<\n",
      "[20:07:09] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC126M3N22.fasta' contents.\n",
      "[20:07:09] Setting --locustag IBFBDDFI from MD5 2bfbddf2e61f115c0489be2a38a06e9e\n",
      "[20:07:09] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22\n",
      "[20:07:09] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\n",
      "[20:07:09] Using filename prefix: pEC126M3N22.XXX\n",
      "[20:07:09] Setting HMMER_NCPU=1\n",
      "[20:07:09] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22/pEC126M3N22.log\n",
      "[20:07:09] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22 --prefix pEC126M3N22 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC126M3N22.fasta\n",
      "[20:07:09] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:07:09] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:07:09] Determined aragorn version is 1.2\n",
      "[20:07:09] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:07:09] Determined barrnap version is 0.9\n",
      "[20:07:09] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:07:09] Determined blastp version is 2.9\n",
      "[20:07:09] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:07:09] Determined cmpress version is 1.1\n",
      "[20:07:09] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:07:09] Determined cmscan version is 1.1\n",
      "[20:07:09] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:07:09] Looking for 'find' - found /usr/bin/find\n",
      "[20:07:09] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:07:09] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:07:09] Determined hmmpress version is 3.3\n",
      "[20:07:09] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:07:09] Determined hmmscan version is 3.3\n",
      "[20:07:09] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:07:09] Looking for 'less' - found /usr/bin/less\n",
      "[20:07:09] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:07:10] Determined makeblastdb version is 2.9\n",
      "[20:07:10] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:07:10] Determined minced version is 4.2\n",
      "[20:07:10] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:07:10] Determined parallel version is 20160622\n",
      "[20:07:10] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:07:10] Determined prodigal version is 2.6\n",
      "[20:07:10] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:07:10] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:07:10] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:07:10] Determined tbl2asn version is 25.7\n",
      "[20:07:10] Using genetic code table 11.\n",
      "[20:07:10] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC126M3N22.fasta\n",
      "[20:07:10] Wrote 1 contigs totalling 73939 bp.\n",
      "[20:07:10] Predicting tRNAs and tmRNAs\n",
      "[20:07:10] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/pEC126M3N22\\.fna\n",
      "[20:07:10] Found 0 tRNAs\n",
      "[20:07:10] Predicting Ribosomal RNAs\n",
      "[20:07:10] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22/pEC126M3N22.fna\n",
      "[20:07:10] Found 0 rRNAs\n",
      "[20:07:10] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:07:10] Total of 0 tRNA + rRNA features\n",
      "[20:07:10] Searching for CRISPR repeats\n",
      "[20:07:10] Found 0 CRISPRs\n",
      "[20:07:10] Predicting coding sequences\n",
      "[20:07:10] Contigs total 73939 bp, so using meta mode\n",
      "[20:07:10] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/pEC126M3N22\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:07:12] Found 75 CDS\n",
      "[20:07:12] Connecting features back to sequences\n",
      "[20:07:12] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:07:12] Annotating CDS, please be patient.\n",
      "[20:07:12] Will use 8 CPUs for similarity searching.\n",
      "[20:07:12] There are still 75 unannotated CDS left (started with 75)\n",
      "[20:07:12] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:07:12] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 1353 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/sprot\\.blast 2> /dev/null\n",
      "[20:07:15] Modify product: Uncharacterized protein YggR => putative protein YggR\n",
      "[20:07:15] Cleaned 1 /product names\n",
      "[20:07:15] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22/sprot.faa\n",
      "[20:07:15] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22/sprot.blast\n",
      "[20:07:15] There are still 58 unannotated CDS left (started with 75)\n",
      "[20:07:15] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:07:15] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 927 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:07:16] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22/HAMAP.hmm.faa\n",
      "[20:07:16] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22/HAMAP.hmm.hmmer3\n",
      "[20:07:16] Labelling remaining 55 proteins as 'hypothetical protein'\n",
      "[20:07:16] Found 20 unique /gene codes.\n",
      "[20:07:16] Fixed 0 colliding /gene names.\n",
      "[20:07:16] Adding /locus_tag identifiers\n",
      "[20:07:16] Assigned 75 locus_tags to CDS and RNA features.\n",
      "[20:07:16] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22/\n",
      "[20:07:16] Generating annotation statistics file\n",
      "[20:07:16] Generating Genbank and Sequin files\n",
      "[20:07:16] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/pEC126M3N22\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/pEC126M3N22\\.fsa 2> /dev/null\n",
      "[20:07:16] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/pEC126M3N22\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC126M3N22\\/pEC126M3N22\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC126M3N22 --prefix pEC126M3N22 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC126M3N22.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:07:26] This is prokka 1.13\n",
      "[20:07:26] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:07:26] Homepage is https://github.com/tseemann/prokka\n",
      "[20:07:26] Local time is Wed Jun 24 20:07:26 2020\n",
      "[20:07:26] You are tiago\n",
      "[20:07:26] Operating system is linux\n",
      "[20:07:26] You have BioPerl 1.007002\n",
      "[20:07:26] System has 12 cores.\n",
      "[20:07:26] Will use maximum of 8 cores.\n",
      "[20:07:26] Annotating as >>> Bacteria <<<\n",
      "[20:07:26] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC537BN37.fasta' contents.\n",
      "[20:07:26] Setting --locustag CAFGOAIM from MD5 caf08a269d86bc6c5d965155b3f2280d\n",
      "[20:07:26] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37\n",
      "[20:07:26] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\n",
      "[20:07:26] Using filename prefix: pEC537BN37.XXX\n",
      "[20:07:26] Setting HMMER_NCPU=1\n",
      "[20:07:26] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37/pEC537BN37.log\n",
      "[20:07:26] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37 --prefix pEC537BN37 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC537BN37.fasta\n",
      "[20:07:26] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:07:26] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:07:26] Determined aragorn version is 1.2\n",
      "[20:07:26] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:07:26] Determined barrnap version is 0.9\n",
      "[20:07:26] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:07:26] Determined blastp version is 2.9\n",
      "[20:07:26] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:07:26] Determined cmpress version is 1.1\n",
      "[20:07:26] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:07:26] Determined cmscan version is 1.1\n",
      "[20:07:26] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:07:26] Looking for 'find' - found /usr/bin/find\n",
      "[20:07:26] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:07:26] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:07:26] Determined hmmpress version is 3.3\n",
      "[20:07:26] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:07:26] Determined hmmscan version is 3.3\n",
      "[20:07:26] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:07:26] Looking for 'less' - found /usr/bin/less\n",
      "[20:07:26] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:07:26] Determined makeblastdb version is 2.9\n",
      "[20:07:26] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:07:26] Determined minced version is 4.2\n",
      "[20:07:26] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:07:26] Determined parallel version is 20160622\n",
      "[20:07:26] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:07:26] Determined prodigal version is 2.6\n",
      "[20:07:26] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:07:26] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:07:26] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:07:26] Determined tbl2asn version is 25.7\n",
      "[20:07:26] Using genetic code table 11.\n",
      "[20:07:26] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC537BN37.fasta\n",
      "[20:07:26] Wrote 1 contigs totalling 43294 bp.\n",
      "[20:07:26] Predicting tRNAs and tmRNAs\n",
      "[20:07:26] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/pEC537BN37\\.fna\n",
      "[20:07:26] Found 0 tRNAs\n",
      "[20:07:26] Predicting Ribosomal RNAs\n",
      "[20:07:26] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37/pEC537BN37.fna\n",
      "[20:07:27] Found 0 rRNAs\n",
      "[20:07:27] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:07:27] Total of 0 tRNA + rRNA features\n",
      "[20:07:27] Searching for CRISPR repeats\n",
      "[20:07:27] Found 0 CRISPRs\n",
      "[20:07:27] Predicting coding sequences\n",
      "[20:07:27] Contigs total 43294 bp, so using meta mode\n",
      "[20:07:27] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/pEC537BN37\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:07:28] Found 53 CDS\n",
      "[20:07:28] Connecting features back to sequences\n",
      "[20:07:28] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:07:28] Annotating CDS, please be patient.\n",
      "[20:07:28] Will use 8 CPUs for similarity searching.\n",
      "[20:07:28] There are still 53 unannotated CDS left (started with 53)\n",
      "[20:07:28] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:07:28] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 724 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/sprot\\.blast 2> /dev/null\n",
      "[20:07:29] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37/sprot.faa\n",
      "[20:07:29] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37/sprot.blast\n",
      "[20:07:29] There are still 36 unannotated CDS left (started with 53)\n",
      "[20:07:29] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:07:29] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 406 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:07:30] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37/HAMAP.hmm.faa\n",
      "[20:07:30] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37/HAMAP.hmm.hmmer3\n",
      "[20:07:30] Labelling remaining 34 proteins as 'hypothetical protein'\n",
      "[20:07:30] Found 18 unique /gene codes.\n",
      "[20:07:30] Fixed 2 duplicate /gene - cia_1 cia_2\n",
      "[20:07:30] Fixed 1 colliding /gene names.\n",
      "[20:07:30] Adding /locus_tag identifiers\n",
      "[20:07:30] Assigned 53 locus_tags to CDS and RNA features.\n",
      "[20:07:30] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37/\n",
      "[20:07:30] Generating annotation statistics file\n",
      "[20:07:30] Generating Genbank and Sequin files\n",
      "[20:07:30] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/pEC537BN37\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/pEC537BN37\\.fsa 2> /dev/null\n",
      "[20:07:31] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/pEC537BN37\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pEC537BN37\\/pEC537BN37\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pEC537BN37 --prefix pEC537BN37 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pEC537BN37.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:07:31] This is prokka 1.13\n",
      "[20:07:31] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:07:31] Homepage is https://github.com/tseemann/prokka\n",
      "[20:07:31] Local time is Wed Jun 24 20:07:31 2020\n",
      "[20:07:31] You are tiago\n",
      "[20:07:31] Operating system is linux\n",
      "[20:07:31] You have BioPerl 1.007002\n",
      "[20:07:31] System has 12 cores.\n",
      "[20:07:31] Will use maximum of 8 cores.\n",
      "[20:07:31] Annotating as >>> Bacteria <<<\n",
      "[20:07:31] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP125M3N44.fasta' contents.\n",
      "[20:07:31] Setting --locustag LONAPIGH from MD5 587a9201a21a020bfc6878d460da3ddf\n",
      "[20:07:31] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44\n",
      "[20:07:31] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\n",
      "[20:07:31] Using filename prefix: pKP125M3N44.XXX\n",
      "[20:07:31] Setting HMMER_NCPU=1\n",
      "[20:07:31] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44/pKP125M3N44.log\n",
      "[20:07:31] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44 --prefix pKP125M3N44 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP125M3N44.fasta\n",
      "[20:07:31] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:07:31] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:07:31] Determined aragorn version is 1.2\n",
      "[20:07:31] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:07:31] Determined barrnap version is 0.9\n",
      "[20:07:31] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:07:31] Determined blastp version is 2.9\n",
      "[20:07:31] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:07:31] Determined cmpress version is 1.1\n",
      "[20:07:31] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:07:31] Determined cmscan version is 1.1\n",
      "[20:07:31] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:07:31] Looking for 'find' - found /usr/bin/find\n",
      "[20:07:31] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:07:31] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:07:31] Determined hmmpress version is 3.3\n",
      "[20:07:31] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:07:31] Determined hmmscan version is 3.3\n",
      "[20:07:31] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:07:31] Looking for 'less' - found /usr/bin/less\n",
      "[20:07:31] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:07:31] Determined makeblastdb version is 2.9\n",
      "[20:07:31] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:07:31] Determined minced version is 4.2\n",
      "[20:07:31] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:07:32] Determined parallel version is 20160622\n",
      "[20:07:32] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:07:32] Determined prodigal version is 2.6\n",
      "[20:07:32] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:07:32] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:07:32] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:07:32] Determined tbl2asn version is 25.7\n",
      "[20:07:32] Using genetic code table 11.\n",
      "[20:07:32] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP125M3N44.fasta\n",
      "[20:07:32] Wrote 1 contigs totalling 43061 bp.\n",
      "[20:07:32] Predicting tRNAs and tmRNAs\n",
      "[20:07:32] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/pKP125M3N44\\.fna\n",
      "[20:07:32] Found 0 tRNAs\n",
      "[20:07:32] Predicting Ribosomal RNAs\n",
      "[20:07:32] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44/pKP125M3N44.fna\n",
      "[20:07:32] Found 0 rRNAs\n",
      "[20:07:32] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:07:32] Total of 0 tRNA + rRNA features\n",
      "[20:07:32] Searching for CRISPR repeats\n",
      "[20:07:32] Found 0 CRISPRs\n",
      "[20:07:32] Predicting coding sequences\n",
      "[20:07:32] Contigs total 43061 bp, so using meta mode\n",
      "[20:07:32] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/pKP125M3N44\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:07:33] Found 60 CDS\n",
      "[20:07:33] Connecting features back to sequences\n",
      "[20:07:33] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:07:33] Annotating CDS, please be patient.\n",
      "[20:07:33] Will use 8 CPUs for similarity searching.\n",
      "[20:07:33] There are still 60 unannotated CDS left (started with 60)\n",
      "[20:07:33] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:07:33] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 748 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/sprot\\.blast 2> /dev/null\n",
      "[20:07:34] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44/sprot.faa\n",
      "[20:07:34] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44/sprot.blast\n",
      "[20:07:34] There are still 42 unannotated CDS left (started with 60)\n",
      "[20:07:34] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:07:34] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 380 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:07:35] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44/HAMAP.hmm.faa\n",
      "[20:07:35] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44/HAMAP.hmm.hmmer3\n",
      "[20:07:35] Labelling remaining 39 proteins as 'hypothetical protein'\n",
      "[20:07:35] Found 19 unique /gene codes.\n",
      "[20:07:35] Fixed 0 colliding /gene names.\n",
      "[20:07:35] Adding /locus_tag identifiers\n",
      "[20:07:35] Assigned 60 locus_tags to CDS and RNA features.\n",
      "[20:07:35] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44/\n",
      "[20:07:35] Generating annotation statistics file\n",
      "[20:07:35] Generating Genbank and Sequin files\n",
      "[20:07:35] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/pKP125M3N44\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/pKP125M3N44\\.fsa 2> /dev/null\n",
      "[20:07:36] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/pKP125M3N44\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP125M3N44\\/pKP125M3N44\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP125M3N44 --prefix pKP125M3N44 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP125M3N44.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:07:36] This is prokka 1.13\n",
      "[20:07:36] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:07:36] Homepage is https://github.com/tseemann/prokka\n",
      "[20:07:36] Local time is Wed Jun 24 20:07:36 2020\n",
      "[20:07:36] You are tiago\n",
      "[20:07:36] Operating system is linux\n",
      "[20:07:36] You have BioPerl 1.007002\n",
      "[20:07:36] System has 12 cores.\n",
      "[20:07:36] Will use maximum of 8 cores.\n",
      "[20:07:36] Annotating as >>> Bacteria <<<\n",
      "[20:07:36] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN15.fasta' contents.\n",
      "[20:07:36] Setting --locustag KLAOMLMB from MD5 45a8656b9ae84c356b705e53150c7b36\n",
      "[20:07:36] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15\n",
      "[20:07:36] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\n",
      "[20:07:36] Using filename prefix: pKP508BN15.XXX\n",
      "[20:07:36] Setting HMMER_NCPU=1\n",
      "[20:07:36] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15/pKP508BN15.log\n",
      "[20:07:36] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15 --prefix pKP508BN15 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN15.fasta\n",
      "[20:07:36] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:07:36] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:07:36] Determined aragorn version is 1.2\n",
      "[20:07:36] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:07:36] Determined barrnap version is 0.9\n",
      "[20:07:36] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:07:36] Determined blastp version is 2.9\n",
      "[20:07:36] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:07:36] Determined cmpress version is 1.1\n",
      "[20:07:36] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:07:36] Determined cmscan version is 1.1\n",
      "[20:07:36] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:07:36] Looking for 'find' - found /usr/bin/find\n",
      "[20:07:36] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:07:36] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:07:36] Determined hmmpress version is 3.3\n",
      "[20:07:36] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:07:36] Determined hmmscan version is 3.3\n",
      "[20:07:36] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:07:36] Looking for 'less' - found /usr/bin/less\n",
      "[20:07:36] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:07:36] Determined makeblastdb version is 2.9\n",
      "[20:07:36] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:07:37] Determined minced version is 4.2\n",
      "[20:07:37] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:07:37] Determined parallel version is 20160622\n",
      "[20:07:37] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:07:37] Determined prodigal version is 2.6\n",
      "[20:07:37] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:07:37] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:07:37] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:07:37] Determined tbl2asn version is 25.7\n",
      "[20:07:37] Using genetic code table 11.\n",
      "[20:07:37] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN15.fasta\n",
      "[20:07:37] Wrote 1 contigs totalling 136795 bp.\n",
      "[20:07:37] Predicting tRNAs and tmRNAs\n",
      "[20:07:37] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/pKP508BN15\\.fna\n",
      "[20:07:37] Found 0 tRNAs\n",
      "[20:07:37] Predicting Ribosomal RNAs\n",
      "[20:07:37] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15/pKP508BN15.fna\n",
      "[20:07:37] Found 0 rRNAs\n",
      "[20:07:37] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:07:37] Total of 0 tRNA + rRNA features\n",
      "[20:07:37] Searching for CRISPR repeats\n",
      "[20:07:37] Found 0 CRISPRs\n",
      "[20:07:37] Predicting coding sequences\n",
      "[20:07:37] Contigs total 136795 bp, so using single mode\n",
      "[20:07:37] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/pKP508BN15\\.fna -c -m -g 11 -p single -f sco -q\n",
      "[20:07:38] Found 177 CDS\n",
      "[20:07:38] Connecting features back to sequences\n",
      "[20:07:38] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:07:38] Annotating CDS, please be patient.\n",
      "[20:07:38] Will use 8 CPUs for similarity searching.\n",
      "[20:07:38] There are still 177 unannotated CDS left (started with 177)\n",
      "[20:07:38] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:07:38] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 2630 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/sprot\\.blast 2> /dev/null\n",
      "[20:07:42] Modify product: Uncharacterized protein YcbX => putative protein YcbX\n",
      "[20:07:42] Modify product: Probable BsuMI modification methylase subunit YdiO => putative BsuMI modification methylase subunit YdiO\n",
      "[20:07:42] Cleaned 2 /product names\n",
      "[20:07:42] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15/sprot.faa\n",
      "[20:07:42] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15/sprot.blast\n",
      "[20:07:42] There are still 148 unannotated CDS left (started with 177)\n",
      "[20:07:42] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:07:42] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 2025 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:07:44] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15/HAMAP.hmm.faa\n",
      "[20:07:44] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15/HAMAP.hmm.hmmer3\n",
      "[20:07:44] Labelling remaining 144 proteins as 'hypothetical protein'\n",
      "[20:07:44] Found 33 unique /gene codes.\n",
      "[20:07:44] Fixed 0 colliding /gene names.\n",
      "[20:07:44] Adding /locus_tag identifiers\n",
      "[20:07:44] Assigned 177 locus_tags to CDS and RNA features.\n",
      "[20:07:44] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15/\n",
      "[20:07:44] Generating annotation statistics file\n",
      "[20:07:44] Generating Genbank and Sequin files\n",
      "[20:07:44] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/pKP508BN15\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/pKP508BN15\\.fsa 2> /dev/null\n",
      "[20:07:45] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/pKP508BN15\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN15\\/pKP508BN15\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN15 --prefix pKP508BN15 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN15.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:07:45] This is prokka 1.13\n",
      "[20:07:45] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:07:45] Homepage is https://github.com/tseemann/prokka\n",
      "[20:07:45] Local time is Wed Jun 24 20:07:45 2020\n",
      "[20:07:45] You are tiago\n",
      "[20:07:45] Operating system is linux\n",
      "[20:07:45] You have BioPerl 1.007002\n",
      "[20:07:45] System has 12 cores.\n",
      "[20:07:45] Will use maximum of 8 cores.\n",
      "[20:07:45] Annotating as >>> Bacteria <<<\n",
      "[20:07:45] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN34.fasta' contents.\n",
      "[20:07:45] Setting --locustag EENIKJPO from MD5 ee72439815f63a10b8ab256c41137c19\n",
      "[20:07:45] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34\n",
      "[20:07:45] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\n",
      "[20:07:45] Using filename prefix: pKP508BN34.XXX\n",
      "[20:07:45] Setting HMMER_NCPU=1\n",
      "[20:07:45] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34/pKP508BN34.log\n",
      "[20:07:45] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34 --prefix pKP508BN34 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN34.fasta\n",
      "[20:07:45] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:07:45] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:07:45] Determined aragorn version is 1.2\n",
      "[20:07:45] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:07:45] Determined barrnap version is 0.9\n",
      "[20:07:45] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:07:45] Determined blastp version is 2.9\n",
      "[20:07:45] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:07:45] Determined cmpress version is 1.1\n",
      "[20:07:45] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:07:45] Determined cmscan version is 1.1\n",
      "[20:07:45] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:07:45] Looking for 'find' - found /usr/bin/find\n",
      "[20:07:45] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:07:45] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:07:45] Determined hmmpress version is 3.3\n",
      "[20:07:45] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:07:45] Determined hmmscan version is 3.3\n",
      "[20:07:45] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:07:45] Looking for 'less' - found /usr/bin/less\n",
      "[20:07:45] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:07:45] Determined makeblastdb version is 2.9\n",
      "[20:07:45] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:07:46] Determined minced version is 4.2\n",
      "[20:07:46] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:07:46] Determined parallel version is 20160622\n",
      "[20:07:46] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:07:46] Determined prodigal version is 2.6\n",
      "[20:07:46] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:07:46] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:07:46] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:07:46] Determined tbl2asn version is 25.7\n",
      "[20:07:46] Using genetic code table 11.\n",
      "[20:07:46] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN34.fasta\n",
      "[20:07:46] Wrote 1 contigs totalling 63067 bp.\n",
      "[20:07:46] Predicting tRNAs and tmRNAs\n",
      "[20:07:46] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/pKP508BN34\\.fna\n",
      "[20:07:46] Found 0 tRNAs\n",
      "[20:07:46] Predicting Ribosomal RNAs\n",
      "[20:07:46] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34/pKP508BN34.fna\n",
      "[20:07:46] Found 0 rRNAs\n",
      "[20:07:46] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:07:46] Total of 0 tRNA + rRNA features\n",
      "[20:07:46] Searching for CRISPR repeats\n",
      "[20:07:46] Found 0 CRISPRs\n",
      "[20:07:46] Predicting coding sequences\n",
      "[20:07:46] Contigs total 63067 bp, so using meta mode\n",
      "[20:07:46] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/pKP508BN34\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:07:48] Found 78 CDS\n",
      "[20:07:48] Connecting features back to sequences\n",
      "[20:07:48] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:07:48] Annotating CDS, please be patient.\n",
      "[20:07:48] Will use 8 CPUs for similarity searching.\n",
      "[20:07:48] There are still 78 unannotated CDS left (started with 78)\n",
      "[20:07:48] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:07:48] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 1154 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/sprot\\.blast 2> /dev/null\n",
      "[20:07:50] Modify product: Uncharacterized protein YggR => putative protein YggR\n",
      "[20:07:50] Modify product: UPF0758 protein CT0611 => hypothetical protein\n",
      "[20:07:50] Modify product: Pentapeptide repeat protein VPA0095 => Pentapeptide repeat protein\n",
      "[20:07:50] Cleaned 3 /product names\n",
      "[20:07:50] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34/sprot.faa\n",
      "[20:07:50] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34/sprot.blast\n",
      "[20:07:50] There are still 59 unannotated CDS left (started with 78)\n",
      "[20:07:50] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:07:50] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 804 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:07:51] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34/HAMAP.hmm.faa\n",
      "[20:07:51] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34/HAMAP.hmm.hmmer3\n",
      "[20:07:51] Labelling remaining 59 proteins as 'hypothetical protein'\n",
      "[20:07:51] Found 16 unique /gene codes.\n",
      "[20:07:51] Fixed 2 duplicate /gene - hin_1 hin_2\n",
      "[20:07:51] Fixed 1 colliding /gene names.\n",
      "[20:07:51] Adding /locus_tag identifiers\n",
      "[20:07:51] Assigned 78 locus_tags to CDS and RNA features.\n",
      "[20:07:51] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34/\n",
      "[20:07:52] Generating annotation statistics file\n",
      "[20:07:52] Generating Genbank and Sequin files\n",
      "[20:07:52] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/pKP508BN34\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/pKP508BN34\\.fsa 2> /dev/null\n",
      "[20:07:52] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/pKP508BN34\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP508BN34\\/pKP508BN34\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP508BN34 --prefix pKP508BN34 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP508BN34.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:07:52] This is prokka 1.13\n",
      "[20:07:52] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:07:52] Homepage is https://github.com/tseemann/prokka\n",
      "[20:07:52] Local time is Wed Jun 24 20:07:52 2020\n",
      "[20:07:52] You are tiago\n",
      "[20:07:52] Operating system is linux\n",
      "[20:07:52] You have BioPerl 1.007002\n",
      "[20:07:52] System has 12 cores.\n",
      "[20:07:52] Will use maximum of 8 cores.\n",
      "[20:07:52] Annotating as >>> Bacteria <<<\n",
      "[20:07:52] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP98M3N42.fasta' contents.\n",
      "[20:07:52] Setting --locustag FEDIEKEO from MD5 fed2e4e8f247f87e9c978057f9da0876\n",
      "[20:07:52] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42\n",
      "[20:07:52] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\n",
      "[20:07:52] Using filename prefix: pKP98M3N42.XXX\n",
      "[20:07:52] Setting HMMER_NCPU=1\n",
      "[20:07:52] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42/pKP98M3N42.log\n",
      "[20:07:52] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42 --prefix pKP98M3N42 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP98M3N42.fasta\n",
      "[20:07:52] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:07:52] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:07:52] Determined aragorn version is 1.2\n",
      "[20:07:52] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:07:52] Determined barrnap version is 0.9\n",
      "[20:07:52] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:07:53] Determined blastp version is 2.9\n",
      "[20:07:53] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:07:53] Determined cmpress version is 1.1\n",
      "[20:07:53] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:07:53] Determined cmscan version is 1.1\n",
      "[20:07:53] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:07:53] Looking for 'find' - found /usr/bin/find\n",
      "[20:07:53] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:07:53] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:07:53] Determined hmmpress version is 3.3\n",
      "[20:07:53] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:07:53] Determined hmmscan version is 3.3\n",
      "[20:07:53] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:07:53] Looking for 'less' - found /usr/bin/less\n",
      "[20:07:53] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:07:53] Determined makeblastdb version is 2.9\n",
      "[20:07:53] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:07:53] Determined minced version is 4.2\n",
      "[20:07:53] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:07:53] Determined parallel version is 20160622\n",
      "[20:07:53] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:07:53] Determined prodigal version is 2.6\n",
      "[20:07:53] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:07:53] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:07:53] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:07:53] Determined tbl2asn version is 25.7\n",
      "[20:07:53] Using genetic code table 11.\n",
      "[20:07:53] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP98M3N42.fasta\n",
      "[20:07:53] Wrote 1 contigs totalling 42978 bp.\n",
      "[20:07:53] Predicting tRNAs and tmRNAs\n",
      "[20:07:53] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/pKP98M3N42\\.fna\n",
      "[20:07:53] Found 0 tRNAs\n",
      "[20:07:53] Predicting Ribosomal RNAs\n",
      "[20:07:53] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42/pKP98M3N42.fna\n",
      "[20:07:53] Found 0 rRNAs\n",
      "[20:07:53] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:07:53] Total of 0 tRNA + rRNA features\n",
      "[20:07:53] Searching for CRISPR repeats\n",
      "[20:07:53] Found 0 CRISPRs\n",
      "[20:07:53] Predicting coding sequences\n",
      "[20:07:53] Contigs total 42978 bp, so using meta mode\n",
      "[20:07:53] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/pKP98M3N42\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:07:54] Found 60 CDS\n",
      "[20:07:54] Connecting features back to sequences\n",
      "[20:07:54] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:07:54] Annotating CDS, please be patient.\n",
      "[20:07:54] Will use 8 CPUs for similarity searching.\n",
      "[20:07:54] There are still 60 unannotated CDS left (started with 60)\n",
      "[20:07:54] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:07:54] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 748 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/sprot\\.blast 2> /dev/null\n",
      "[20:07:56] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42/sprot.faa\n",
      "[20:07:56] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42/sprot.blast\n",
      "[20:07:56] There are still 42 unannotated CDS left (started with 60)\n",
      "[20:07:56] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:07:56] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 380 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:07:57] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42/HAMAP.hmm.faa\n",
      "[20:07:57] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42/HAMAP.hmm.hmmer3\n",
      "[20:07:57] Labelling remaining 39 proteins as 'hypothetical protein'\n",
      "[20:07:57] Found 19 unique /gene codes.\n",
      "[20:07:57] Fixed 0 colliding /gene names.\n",
      "[20:07:57] Adding /locus_tag identifiers\n",
      "[20:07:57] Assigned 60 locus_tags to CDS and RNA features.\n",
      "[20:07:57] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42/\n",
      "[20:07:57] Generating annotation statistics file\n",
      "[20:07:57] Generating Genbank and Sequin files\n",
      "[20:07:57] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/pKP98M3N42\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/pKP98M3N42\\.fsa 2> /dev/null\n",
      "[20:07:57] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/pKP98M3N42\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pKP98M3N42\\/pKP98M3N42\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pKP98M3N42 --prefix pKP98M3N42 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pKP98M3N42.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:07:57] This is prokka 1.13\n",
      "[20:07:57] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:07:57] Homepage is https://github.com/tseemann/prokka\n",
      "[20:07:57] Local time is Wed Jun 24 20:07:57 2020\n",
      "[20:07:57] You are tiago\n",
      "[20:07:57] Operating system is linux\n",
      "[20:07:57] You have BioPerl 1.007002\n",
      "[20:07:57] System has 12 cores.\n",
      "[20:07:57] Will use maximum of 8 cores.\n",
      "[20:07:57] Annotating as >>> Bacteria <<<\n",
      "[20:07:57] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE452BN56.fasta' contents.\n",
      "[20:07:57] Setting --locustag CFNJLOCI from MD5 cf7358c223d204131098bf8adec771e3\n",
      "[20:07:57] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56\n",
      "[20:07:57] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\n",
      "[20:07:57] Using filename prefix: pSE452BN56.XXX\n",
      "[20:07:57] Setting HMMER_NCPU=1\n",
      "[20:07:57] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56/pSE452BN56.log\n",
      "[20:07:57] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56 --prefix pSE452BN56 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE452BN56.fasta\n",
      "[20:07:57] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:07:57] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:07:57] Determined aragorn version is 1.2\n",
      "[20:07:57] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:07:58] Determined barrnap version is 0.9\n",
      "[20:07:58] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:07:58] Determined blastp version is 2.9\n",
      "[20:07:58] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:07:58] Determined cmpress version is 1.1\n",
      "[20:07:58] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:07:58] Determined cmscan version is 1.1\n",
      "[20:07:58] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:07:58] Looking for 'find' - found /usr/bin/find\n",
      "[20:07:58] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:07:58] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:07:58] Determined hmmpress version is 3.3\n",
      "[20:07:58] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:07:58] Determined hmmscan version is 3.3\n",
      "[20:07:58] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:07:58] Looking for 'less' - found /usr/bin/less\n",
      "[20:07:58] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:07:58] Determined makeblastdb version is 2.9\n",
      "[20:07:58] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:07:58] Determined minced version is 4.2\n",
      "[20:07:58] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:07:58] Determined parallel version is 20160622\n",
      "[20:07:58] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:07:58] Determined prodigal version is 2.6\n",
      "[20:07:58] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:07:58] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:07:58] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:07:58] Determined tbl2asn version is 25.7\n",
      "[20:07:58] Using genetic code table 11.\n",
      "[20:07:58] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE452BN56.fasta\n",
      "[20:07:58] Wrote 1 contigs totalling 2305 bp.\n",
      "[20:07:58] Predicting tRNAs and tmRNAs\n",
      "[20:07:58] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/pSE452BN56\\.fna\n",
      "[20:07:58] Found 0 tRNAs\n",
      "[20:07:58] Predicting Ribosomal RNAs\n",
      "[20:07:58] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56/pSE452BN56.fna\n",
      "[20:07:58] Found 0 rRNAs\n",
      "[20:07:58] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:07:58] Total of 0 tRNA + rRNA features\n",
      "[20:07:58] Searching for CRISPR repeats\n",
      "[20:07:58] Found 0 CRISPRs\n",
      "[20:07:58] Predicting coding sequences\n",
      "[20:07:58] Contigs total 2305 bp, so using meta mode\n",
      "[20:07:58] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/pSE452BN56\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:07:58] Found 1 CDS\n",
      "[20:07:58] Connecting features back to sequences\n",
      "[20:07:58] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:07:58] Annotating CDS, please be patient.\n",
      "[20:07:58] Will use 8 CPUs for similarity searching.\n",
      "[20:07:58] There are still 1 unannotated CDS left (started with 1)\n",
      "[20:07:58] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:07:58] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 15 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/sprot\\.blast 2> /dev/null\n",
      "[20:07:59] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56/sprot.faa\n",
      "[20:07:59] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56/sprot.blast\n",
      "[20:07:59] Found 1 unique /gene codes.\n",
      "[20:07:59] Fixed 0 colliding /gene names.\n",
      "[20:07:59] Adding /locus_tag identifiers\n",
      "[20:07:59] Assigned 1 locus_tags to CDS and RNA features.\n",
      "[20:07:59] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56/\n",
      "[20:07:59] Generating annotation statistics file\n",
      "[20:07:59] Generating Genbank and Sequin files\n",
      "[20:07:59] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/pSE452BN56\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/pSE452BN56\\.fsa 2> /dev/null\n",
      "[20:07:59] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/pSE452BN56\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE452BN56\\/pSE452BN56\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE452BN56 --prefix pSE452BN56 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE452BN56.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:08:00] This is prokka 1.13\n",
      "[20:08:00] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:08:00] Homepage is https://github.com/tseemann/prokka\n",
      "[20:08:00] Local time is Wed Jun 24 20:08:00 2020\n",
      "[20:08:00] You are tiago\n",
      "[20:08:00] Operating system is linux\n",
      "[20:08:00] You have BioPerl 1.007002\n",
      "[20:08:00] System has 12 cores.\n",
      "[20:08:00] Will use maximum of 8 cores.\n",
      "[20:08:00] Annotating as >>> Bacteria <<<\n",
      "[20:08:00] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE605AN41.fasta' contents.\n",
      "[20:08:00] Setting --locustag OPMNFDHN from MD5 8967fd173e3569682287945dba21b024\n",
      "[20:08:00] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41\n",
      "[20:08:00] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\n",
      "[20:08:00] Using filename prefix: pSE605AN41.XXX\n",
      "[20:08:00] Setting HMMER_NCPU=1\n",
      "[20:08:00] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41/pSE605AN41.log\n",
      "[20:08:00] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41 --prefix pSE605AN41 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE605AN41.fasta\n",
      "[20:08:00] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:08:00] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:08:00] Determined aragorn version is 1.2\n",
      "[20:08:00] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:08:00] Determined barrnap version is 0.9\n",
      "[20:08:00] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:08:00] Determined blastp version is 2.9\n",
      "[20:08:00] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:08:00] Determined cmpress version is 1.1\n",
      "[20:08:00] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:08:00] Determined cmscan version is 1.1\n",
      "[20:08:00] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:08:00] Looking for 'find' - found /usr/bin/find\n",
      "[20:08:00] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:08:00] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:08:00] Determined hmmpress version is 3.3\n",
      "[20:08:00] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:08:00] Determined hmmscan version is 3.3\n",
      "[20:08:00] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:08:00] Looking for 'less' - found /usr/bin/less\n",
      "[20:08:00] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:08:00] Determined makeblastdb version is 2.9\n",
      "[20:08:00] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:08:00] Determined minced version is 4.2\n",
      "[20:08:00] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:08:00] Determined parallel version is 20160622\n",
      "[20:08:00] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:08:00] Determined prodigal version is 2.6\n",
      "[20:08:00] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:08:00] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:08:00] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:08:00] Determined tbl2asn version is 25.7\n",
      "[20:08:00] Using genetic code table 11.\n",
      "[20:08:00] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE605AN41.fasta\n",
      "[20:08:00] Wrote 1 contigs totalling 4753 bp.\n",
      "[20:08:00] Predicting tRNAs and tmRNAs\n",
      "[20:08:00] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/pSE605AN41\\.fna\n",
      "[20:08:00] Found 0 tRNAs\n",
      "[20:08:00] Predicting Ribosomal RNAs\n",
      "[20:08:00] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41/pSE605AN41.fna\n",
      "[20:08:01] Found 0 rRNAs\n",
      "[20:08:01] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:08:01] Total of 0 tRNA + rRNA features\n",
      "[20:08:01] Searching for CRISPR repeats\n",
      "[20:08:01] Found 0 CRISPRs\n",
      "[20:08:01] Predicting coding sequences\n",
      "[20:08:01] Contigs total 4753 bp, so using meta mode\n",
      "[20:08:01] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/pSE605AN41\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:08:01] Found 6 CDS\n",
      "[20:08:01] Connecting features back to sequences\n",
      "[20:08:01] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:08:01] Annotating CDS, please be patient.\n",
      "[20:08:01] Will use 8 CPUs for similarity searching.\n",
      "[20:08:01] There are still 6 unannotated CDS left (started with 6)\n",
      "[20:08:01] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:08:01] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 75 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/sprot\\.blast 2> /dev/null\n",
      "[20:08:01] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41/sprot.faa\n",
      "[20:08:01] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41/sprot.blast\n",
      "[20:08:01] There are still 3 unannotated CDS left (started with 6)\n",
      "[20:08:01] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:08:01] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 24 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:08:02] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41/HAMAP.hmm.faa\n",
      "[20:08:02] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41/HAMAP.hmm.hmmer3\n",
      "[20:08:02] Labelling remaining 3 proteins as 'hypothetical protein'\n",
      "[20:08:02] Found 3 unique /gene codes.\n",
      "[20:08:02] Fixed 0 colliding /gene names.\n",
      "[20:08:02] Adding /locus_tag identifiers\n",
      "[20:08:02] Assigned 6 locus_tags to CDS and RNA features.\n",
      "[20:08:02] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41/\n",
      "[20:08:02] Generating annotation statistics file\n",
      "[20:08:02] Generating Genbank and Sequin files\n",
      "[20:08:02] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/pSE605AN41\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/pSE605AN41\\.fsa 2> /dev/null\n",
      "[20:08:02] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/pSE605AN41\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSE605AN41\\/pSE605AN41\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSE605AN41 --prefix pSE605AN41 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSE605AN41.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:08:03] This is prokka 1.13\n",
      "[20:08:03] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:08:03] Homepage is https://github.com/tseemann/prokka\n",
      "[20:08:03] Local time is Wed Jun 24 20:08:03 2020\n",
      "[20:08:03] You are tiago\n",
      "[20:08:03] Operating system is linux\n",
      "[20:08:03] You have BioPerl 1.007002\n",
      "[20:08:03] System has 12 cores.\n",
      "[20:08:03] Will use maximum of 8 cores.\n",
      "[20:08:03] Annotating as >>> Bacteria <<<\n",
      "[20:08:03] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW579AN18.fasta' contents.\n",
      "[20:08:03] Setting --locustag NLAABJDB from MD5 75aab3dbde4a4437829aed46c1fd36dc\n",
      "[20:08:03] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18\n",
      "[20:08:03] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\n",
      "[20:08:03] Using filename prefix: pSW579AN18.XXX\n",
      "[20:08:03] Setting HMMER_NCPU=1\n",
      "[20:08:03] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18/pSW579AN18.log\n",
      "[20:08:03] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18 --prefix pSW579AN18 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW579AN18.fasta\n",
      "[20:08:03] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:08:03] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:08:03] Determined aragorn version is 1.2\n",
      "[20:08:03] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:08:03] Determined barrnap version is 0.9\n",
      "[20:08:03] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:08:03] Determined blastp version is 2.9\n",
      "[20:08:03] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:08:03] Determined cmpress version is 1.1\n",
      "[20:08:03] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:08:03] Determined cmscan version is 1.1\n",
      "[20:08:03] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:08:03] Looking for 'find' - found /usr/bin/find\n",
      "[20:08:03] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:08:03] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:08:03] Determined hmmpress version is 3.3\n",
      "[20:08:03] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:08:03] Determined hmmscan version is 3.3\n",
      "[20:08:03] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:08:03] Looking for 'less' - found /usr/bin/less\n",
      "[20:08:03] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:08:03] Determined makeblastdb version is 2.9\n",
      "[20:08:03] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:08:03] Determined minced version is 4.2\n",
      "[20:08:03] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:08:03] Determined parallel version is 20160622\n",
      "[20:08:03] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:08:03] Determined prodigal version is 2.6\n",
      "[20:08:03] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:08:03] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:08:03] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:08:03] Determined tbl2asn version is 25.7\n",
      "[20:08:03] Using genetic code table 11.\n",
      "[20:08:03] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW579AN18.fasta\n",
      "[20:08:03] Wrote 1 contigs totalling 24824 bp.\n",
      "[20:08:03] Predicting tRNAs and tmRNAs\n",
      "[20:08:03] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/pSW579AN18\\.fna\n",
      "[20:08:03] Found 0 tRNAs\n",
      "[20:08:03] Predicting Ribosomal RNAs\n",
      "[20:08:03] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18/pSW579AN18.fna\n",
      "[20:08:03] Found 0 rRNAs\n",
      "[20:08:03] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:08:03] Total of 0 tRNA + rRNA features\n",
      "[20:08:03] Searching for CRISPR repeats\n",
      "[20:08:04] Found 0 CRISPRs\n",
      "[20:08:04] Predicting coding sequences\n",
      "[20:08:04] Contigs total 24824 bp, so using meta mode\n",
      "[20:08:04] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/pSW579AN18\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:08:04] Found 28 CDS\n",
      "[20:08:04] Connecting features back to sequences\n",
      "[20:08:04] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:08:04] Annotating CDS, please be patient.\n",
      "[20:08:04] Will use 8 CPUs for similarity searching.\n",
      "[20:08:04] There are still 28 unannotated CDS left (started with 28)\n",
      "[20:08:04] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:08:04] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 320 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/sprot\\.blast 2> /dev/null\n",
      "[20:08:05] Modify product: Uncharacterized protein SAV1875 => putative protein\n",
      "[20:08:05] Modify product: Probable malate:quinone oxidoreductase 2 => putative malate:quinone oxidoreductase 2\n",
      "[20:08:05] Cleaned 2 /product names\n",
      "[20:08:05] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18/sprot.faa\n",
      "[20:08:05] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18/sprot.blast\n",
      "[20:08:05] There are still 16 unannotated CDS left (started with 28)\n",
      "[20:08:05] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:08:05] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 130 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:08:06] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18/HAMAP.hmm.faa\n",
      "[20:08:06] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18/HAMAP.hmm.hmmer3\n",
      "[20:08:06] Labelling remaining 15 proteins as 'hypothetical protein'\n",
      "[20:08:06] Found 10 unique /gene codes.\n",
      "[20:08:06] Fixed 0 colliding /gene names.\n",
      "[20:08:06] Adding /locus_tag identifiers\n",
      "[20:08:06] Assigned 28 locus_tags to CDS and RNA features.\n",
      "[20:08:06] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18/\n",
      "[20:08:06] Generating annotation statistics file\n",
      "[20:08:06] Generating Genbank and Sequin files\n",
      "[20:08:06] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/pSW579AN18\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/pSW579AN18\\.fsa 2> /dev/null\n",
      "[20:08:06] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/pSW579AN18\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW579AN18\\/pSW579AN18\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW579AN18 --prefix pSW579AN18 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW579AN18.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:08:06] This is prokka 1.13\n",
      "[20:08:06] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:08:06] Homepage is https://github.com/tseemann/prokka\n",
      "[20:08:06] Local time is Wed Jun 24 20:08:06 2020\n",
      "[20:08:06] You are tiago\n",
      "[20:08:06] Operating system is linux\n",
      "[20:08:06] You have BioPerl 1.007002\n",
      "[20:08:06] System has 12 cores.\n",
      "[20:08:06] Will use maximum of 8 cores.\n",
      "[20:08:06] Annotating as >>> Bacteria <<<\n",
      "[20:08:06] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN27.fasta' contents.\n",
      "[20:08:06] Setting --locustag LLCPLGJP from MD5 55c95039a4f3b75a902042bed2470ccd\n",
      "[20:08:06] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27\n",
      "[20:08:06] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\n",
      "[20:08:06] Using filename prefix: pSW732BN27.XXX\n",
      "[20:08:06] Setting HMMER_NCPU=1\n",
      "[20:08:06] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27/pSW732BN27.log\n",
      "[20:08:06] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27 --prefix pSW732BN27 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN27.fasta\n",
      "[20:08:06] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:08:06] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:08:06] Determined aragorn version is 1.2\n",
      "[20:08:06] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:08:06] Determined barrnap version is 0.9\n",
      "[20:08:06] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:08:06] Determined blastp version is 2.9\n",
      "[20:08:06] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:08:06] Determined cmpress version is 1.1\n",
      "[20:08:06] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:08:06] Determined cmscan version is 1.1\n",
      "[20:08:06] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:08:06] Looking for 'find' - found /usr/bin/find\n",
      "[20:08:06] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:08:06] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:08:06] Determined hmmpress version is 3.3\n",
      "[20:08:06] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:08:06] Determined hmmscan version is 3.3\n",
      "[20:08:06] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:08:06] Looking for 'less' - found /usr/bin/less\n",
      "[20:08:06] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:08:07] Determined makeblastdb version is 2.9\n",
      "[20:08:07] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:08:07] Determined minced version is 4.2\n",
      "[20:08:07] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:08:07] Determined parallel version is 20160622\n",
      "[20:08:07] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:08:07] Determined prodigal version is 2.6\n",
      "[20:08:07] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:08:07] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:08:07] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:08:07] Determined tbl2asn version is 25.7\n",
      "[20:08:07] Using genetic code table 11.\n",
      "[20:08:07] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN27.fasta\n",
      "[20:08:07] Wrote 1 contigs totalling 10506 bp.\n",
      "[20:08:07] Predicting tRNAs and tmRNAs\n",
      "[20:08:07] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/pSW732BN27\\.fna\n",
      "[20:08:07] Found 0 tRNAs\n",
      "[20:08:07] Predicting Ribosomal RNAs\n",
      "[20:08:07] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27/pSW732BN27.fna\n",
      "[20:08:07] Found 0 rRNAs\n",
      "[20:08:07] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:08:07] Total of 0 tRNA + rRNA features\n",
      "[20:08:07] Searching for CRISPR repeats\n",
      "[20:08:07] Found 0 CRISPRs\n",
      "[20:08:07] Predicting coding sequences\n",
      "[20:08:07] Contigs total 10506 bp, so using meta mode\n",
      "[20:08:07] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/pSW732BN27\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:08:07] Found 10 CDS\n",
      "[20:08:07] Connecting features back to sequences\n",
      "[20:08:07] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:08:07] Annotating CDS, please be patient.\n",
      "[20:08:07] Will use 8 CPUs for similarity searching.\n",
      "[20:08:07] There are still 10 unannotated CDS left (started with 10)\n",
      "[20:08:07] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:08:07] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 169 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/sprot\\.blast 2> /dev/null\n",
      "[20:08:08] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27/sprot.faa\n",
      "[20:08:08] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27/sprot.blast\n",
      "[20:08:08] There are still 4 unannotated CDS left (started with 10)\n",
      "[20:08:08] Will use hmmer3 to search against /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[20:08:08] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 53 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiago/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[20:08:09] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27/HAMAP.hmm.faa\n",
      "[20:08:09] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27/HAMAP.hmm.hmmer3\n",
      "[20:08:09] Labelling remaining 3 proteins as 'hypothetical protein'\n",
      "[20:08:09] Found 7 unique /gene codes.\n",
      "[20:08:09] Fixed 0 colliding /gene names.\n",
      "[20:08:09] Adding /locus_tag identifiers\n",
      "[20:08:09] Assigned 10 locus_tags to CDS and RNA features.\n",
      "[20:08:09] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27/\n",
      "[20:08:09] Generating annotation statistics file\n",
      "[20:08:09] Generating Genbank and Sequin files\n",
      "[20:08:09] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/pSW732BN27\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/pSW732BN27\\.fsa 2> /dev/null\n",
      "[20:08:09] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/pSW732BN27\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN27\\/pSW732BN27\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN27 --prefix pSW732BN27 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN27.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[20:08:09] This is prokka 1.13\n",
      "[20:08:09] Written by Torsten Seemann <torsten.seemann@gmail.com>\n",
      "[20:08:09] Homepage is https://github.com/tseemann/prokka\n",
      "[20:08:09] Local time is Wed Jun 24 20:08:09 2020\n",
      "[20:08:09] You are tiago\n",
      "[20:08:09] Operating system is linux\n",
      "[20:08:09] You have BioPerl 1.007002\n",
      "[20:08:09] System has 12 cores.\n",
      "[20:08:09] Will use maximum of 8 cores.\n",
      "[20:08:09] Annotating as >>> Bacteria <<<\n",
      "[20:08:09] Generating locus_tag from '/home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN43.fasta' contents.\n",
      "[20:08:09] Setting --locustag CCDKKEHC from MD5 ccd44e1c4dfeedb64f1331d8535e7c73\n",
      "[20:08:09] Creating new output folder: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43\n",
      "[20:08:09] Running: mkdir -p \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\n",
      "[20:08:09] Using filename prefix: pSW732BN43.XXX\n",
      "[20:08:09] Setting HMMER_NCPU=1\n",
      "[20:08:09] Writing log to: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43/pSW732BN43.log\n",
      "[20:08:09] Command: /home/tiago/anaconda3/bin/prokka --outdir /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43 --prefix pSW732BN43 /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN43.fasta\n",
      "[20:08:09] Appending to PATH: /home/tiago/anaconda3/bin\n",
      "[20:08:09] Looking for 'aragorn' - found /home/tiago/anaconda3/bin/aragorn\n",
      "[20:08:09] Determined aragorn version is 1.2\n",
      "[20:08:09] Looking for 'barrnap' - found /home/tiago/anaconda3/bin/barrnap\n",
      "[20:08:09] Determined barrnap version is 0.9\n",
      "[20:08:09] Looking for 'blastp' - found /home/tiago/anaconda3/bin/blastp\n",
      "[20:08:09] Determined blastp version is 2.9\n",
      "[20:08:09] Looking for 'cmpress' - found /home/tiago/anaconda3/bin/cmpress\n",
      "[20:08:09] Determined cmpress version is 1.1\n",
      "[20:08:09] Looking for 'cmscan' - found /home/tiago/anaconda3/bin/cmscan\n",
      "[20:08:10] Determined cmscan version is 1.1\n",
      "[20:08:10] Looking for 'egrep' - found /usr/bin/egrep\n",
      "[20:08:10] Looking for 'find' - found /usr/bin/find\n",
      "[20:08:10] Looking for 'grep' - found /usr/bin/grep\n",
      "[20:08:10] Looking for 'hmmpress' - found /home/tiago/anaconda3/bin/hmmpress\n",
      "[20:08:10] Determined hmmpress version is 3.3\n",
      "[20:08:10] Looking for 'hmmscan' - found /home/tiago/anaconda3/bin/hmmscan\n",
      "[20:08:10] Determined hmmscan version is 3.3\n",
      "[20:08:10] Looking for 'java' - found /home/tiago/anaconda3/bin/java\n",
      "[20:08:10] Looking for 'less' - found /usr/bin/less\n",
      "[20:08:10] Looking for 'makeblastdb' - found /home/tiago/anaconda3/bin/makeblastdb\n",
      "[20:08:10] Determined makeblastdb version is 2.9\n",
      "[20:08:10] Looking for 'minced' - found /home/tiago/anaconda3/bin/minced\n",
      "[20:08:10] Determined minced version is 4.2\n",
      "[20:08:10] Looking for 'parallel' - found /home/tiago/anaconda3/bin/parallel\n",
      "[20:08:10] Determined parallel version is 20160622\n",
      "[20:08:10] Looking for 'prodigal' - found /home/tiago/anaconda3/bin/prodigal\n",
      "[20:08:10] Determined prodigal version is 2.6\n",
      "[20:08:10] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiago/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[20:08:10] Looking for 'sed' - found /usr/bin/sed\n",
      "[20:08:10] Looking for 'tbl2asn' - found /home/tiago/anaconda3/bin/tbl2asn\n",
      "[tbl2asn] This copy of tbl2asn is more than a year old.  Please download the current version.\n",
      "[20:08:10] Determined tbl2asn version is 25.7\n",
      "[20:08:10] Using genetic code table 11.\n",
      "[20:08:10] Loading and checking input file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN43.fasta\n",
      "[20:08:10] Wrote 1 contigs totalling 2345 bp.\n",
      "[20:08:10] Predicting tRNAs and tmRNAs\n",
      "[20:08:10] Running: aragorn -l -gc11  -w \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/pSW732BN43\\.fna\n",
      "[20:08:10] Found 0 tRNAs\n",
      "[20:08:10] Predicting Ribosomal RNAs\n",
      "[20:08:10] Running Barrnap with 8 threads\n",
      "[barrnap] ERROR: nhmmer failed to run - # Target file:     /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43/pSW732BN43.fna\n",
      "[20:08:10] Found 0 rRNAs\n",
      "[20:08:10] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[20:08:10] Total of 0 tRNA + rRNA features\n",
      "[20:08:10] Searching for CRISPR repeats\n",
      "[20:08:10] Found 0 CRISPRs\n",
      "[20:08:10] Predicting coding sequences\n",
      "[20:08:10] Contigs total 2345 bp, so using meta mode\n",
      "[20:08:10] Running: prodigal -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/pSW732BN43\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[20:08:10] Found 1 CDS\n",
      "[20:08:10] Connecting features back to sequences\n",
      "[20:08:10] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[20:08:10] Annotating CDS, please be patient.\n",
      "[20:08:10] Will use 8 CPUs for similarity searching.\n",
      "[20:08:10] There are still 1 unannotated CDS left (started with 1)\n",
      "[20:08:10] Will use blast to search against /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[20:08:10] Running: cat \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 15 --recstart '>' --pipe blastp -query - -db /home/tiago/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/sprot\\.blast 2> /dev/null\n",
      "[20:08:11] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43/sprot.faa\n",
      "[20:08:11] Deleting unwanted file: /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43/sprot.blast\n",
      "[20:08:11] Found 1 unique /gene codes.\n",
      "[20:08:11] Fixed 0 colliding /gene names.\n",
      "[20:08:11] Adding /locus_tag identifiers\n",
      "[20:08:11] Assigned 1 locus_tags to CDS and RNA features.\n",
      "[20:08:11] Writing outputs to /home/tiago/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43/\n",
      "[20:08:11] Generating annotation statistics file\n",
      "[20:08:11] Generating Genbank and Sequin files\n",
      "[20:08:11] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/pSW732BN43\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/pSW732BN43\\.fsa 2> /dev/null\n",
      "[20:08:11] Could not run command: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.13 from https://github.com/tseemann/prokka' -Z \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/pSW732BN43\\.err -i \\/home\\/tiago\\/documentos\\/github\\/ARG\\-Sentinel\\/suberror\\/plasmids_annot\\/annot\\/pSW732BN43\\/pSW732BN43\\.fsa 2> /dev/null\n"
     ]
    }
   ],
   "source": [
    "!prokka --outdir ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/annot/pSW732BN43 --prefix pSW732BN43 ~/documentos/github/ARG-Sentinel/suberror/plasmids_annot/pSW732BN43.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
